src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts

```diff
@@ -210,17 +210,17 @@ export class SimpleFileDialog {
 		return newOptions;
 	}
 
-	private remoteUriFrom(path: string): URI {
+	private remoteUriFrom(path: string, hintUri?: URI): URI {
 		if (!path.startsWith('\\\\')) {
 			path = path.replace(/\\/g, '/');
 		}
-		const uri: URI = this.scheme === Schemas.file ? URI.file(path) : URI.from({ scheme: this.scheme, path });
-		// If the default scheme is file, then we don't care about the remote authority
-		const authority = uri.scheme === Schemas.file ? undefined : this.remoteAuthority;
+		const uri: URI = this.scheme === Schemas.file ? URI.file(path) : URI.from({ scheme: this.scheme, path, query: hintUri?.query, fragment: hintUri?.fragment });
+		// If the default scheme is file, then we don't care about the remote authority or the hint authority
+		const authority = uri.scheme === Schemas.file ? undefined : this.remoteAuthority ?? hintUri?.authority;
```

Comment:
Can you add parentheses here to indicate the intended evaluation order?

Reply:
@alexr00 I pushed a change. Does it do what you want?

Reply:
Yes, that's exactly it.